docs/DeveloperGuide.adoc

```diff
@@ -767,14 +809,316 @@ Use case ends.
 +
 Use case resumes at step 2.
 
+[discrete]
+=== Use case: Select person
+
+*Precondition: Person in question should be saved with 'add' in addressbook*
+
+*Guarantees: Person in question will be displayed with all fields visible*
+
+*MSS*
+
+1.  User requests to 'find' a specific person by name
+2.  AddressBook shows a list of persons with name entered
+3.  User selects the index of specific person
+4.  AddressBook shows full contact details of person
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. The given index is invalid
++
+[none]
+** 3a1. AddressBook shows an error message
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: History
+
+*Guarantees: Displays list of commands entered by user*
+
+*MSS*
+
+1.  User requests to see command history
+2.  AddressBook lists the commands entered in reverse order
+3.  User views previous commands
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty
++
+Use case ends.
+
+[discrete]
+=== Use case: Help window
+
+*Guarantees: Displays instructions for command entered by user*
+
+*MSS*
+
+1.  User enters 'help' command
+2.  AddressBook shows the instructions of the command entered
+3.  User views help window
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. No command was previously entered
++
+Use case ends.
+
+* 3a. User chooses to copy help example text
++
+[none]
+** 3a1. User switches back to main window and pastes the command
++
+Use case ends.
+
+[discrete]
+=== Use case: Find person
+
+*MSS*
+
+1.  User requests to find one or more person(s) in AddressBook
+2.  AddressBook shows a list of persons with the name(s) specified by user
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. 0 person found
++
+Use case ends.
+
+[discrete]
+=== Use case: Add person
+
+*MSS*
+
+1.  User input command to add person
+2.  AddressBook adds the person
+3.  AddressBook displays message for successfully adding a person along with the person's details
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. User input command wrongly
++
+[none]
+** 1a1. AddressBook shows an invalid format message and displays the right format to user
++
+Use case resumes at step 1.
+
+* 1b. User added a person already present in AddressBook
++
+[none]
+** 1b1. AddressBook shows duplicate person error message
++
+Use case resumes at step 1.
+
+[discrete]
+=== Use case: List person(s) with specific tag
+
+*MSS*
+
+1.  User requests for a list of person(s) with a specified tag
+2.  AddressBook shows a list of person(s) with the tag specified by user
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. 0 person with such tag
++
+Use case ends.
+
+[discrete]
+=== Use case: Hide Selected Contacts
+
+*MSS*
+
+1.  User requests to a contact with the command: hide
```

Comment:
typo? `requests to a contact` 
i guess you meant requests to **hide** a contact?